src/relay/config/config.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import Dict
+
+import toml
+from marshmallow import ValidationError as MarshmallowValidationError
+from toolz import get_in, update_in
+
+from .schema import ConfigSchema
+
+logger = logging.getLogger("config")
+
+
+def load_config(path: str) -> Dict:
+    raw_data = toml.load(path)
+    raw_data = convert_legacy_format(raw_data)
+    return ConfigSchema().load(raw_data)
+
+
+def dump_config(config: Dict, path: str) -> None:
+    with open(path, "w") as file:
+        toml.dump(ConfigSchema().dump(config), file)
+
+
+def dump_default_config(path: str) -> None:
+    dump_config(generate_default_config(), path)
+
+
+def generate_default_config() -> Dict:
+    return ConfigSchema().load({})
+
+
+ValidationError = MarshmallowValidationError
```

Comment:
probably would do the same yes, I did it because I want to make it explicit that I dont want to directly expose the marshmallow exception. Idea is that the marshmallow code does not leak into the code that uses the config.

Reply:
I changed it, as I anyways had issues with mypy understanding it.